flink-python/src/main/java/org/apache/flink/api/common/python/PythonBridgeUtils.java

```diff
@@ -266,6 +278,92 @@ private static Object getPickledBytesFromJavaObject(Object obj, LogicalType data
 		}
 	}
 
+	public static Object getPickledBytesFromJavaObject(Object obj, TypeInformation<?> dataType) throws IOException {
+		Pickler pickler = new Pickler();
+		initialize();
+		if (obj == null) {
+			return new byte[0];
+		} else {
+			if (dataType instanceof SqlTimeTypeInfo) {
+				SqlTimeTypeInfo<?> sqlTimeTypeInfo = SqlTimeTypeInfo.getInfoFor(dataType.getTypeClass());
+				if (sqlTimeTypeInfo == DATE) {
+					long time;
+					if (obj instanceof LocalDate) {
+						time = ((LocalDate) (obj)).toEpochDay();
+					} else {
+						time = ((Date) obj).toLocalDate().toEpochDay();
```

Comment:
Two thoughts here:
- Isn't it always be Date? 
- Why not just pickle the Date object? I assume pickle could handle the Date/Time/Timestamp objects properly.

Reply:
@dianfu , for `DateType` and `DATE`, the object could both be pickled directly, but for `DateType` this has the type check to pickle the Date object. IMO, this should be unified for `DateType` and `DATE`.